crates/ra_hir_def/src/nameres/collector.rs

```diff
@@ -478,8 +499,17 @@ where
         }
         let scope = &mut self.def_map.modules[module_id].scope;
         let mut changed = false;
-        for (name, res) in resolutions {
-            changed |= scope.push_res(name.clone(), res.with_visibility(vis));
+        for (name, res, is_named) in resolutions {
+            if *is_named {
+                changed |= scope.push_res(name.clone(), res.with_visibility(vis));
+            } else {
+                if let Some(def) = res.take_types() {
+                    match def {
+                        ModuleDefId::TraitId(t) => scope.define_unnamed_trait(t, vis),
```

Comment:
you need to update `changed` here as well (and actually probably check whether the trait was newly added to the scope)